internal/controller/promotions/promotions.go

```diff
@@ -439,23 +433,34 @@ func (r *reconciler) promote(
 		// so we are safe from race conditions and can just update the status
 		// TODO: remove all patching of Stage status out of promo reconciler
 		if err = kubeclient.PatchStatus(ctx, r.kargoClient, stage, func(status *kargoapi.StageStatus) {
-			// control-flow Stage history is maintained in Stage controller.
-			// So we only modify history for normal Stages.
-			// (Technically, we should prevent creating promotion jobs on
-			// control-flow stages in the first place)
```

Comment:
I'm removing this comment, which seems as if it was somewhat misplaced and just opening an issue for it instead.

Reply:
Created #1830